designs/manager_policy_mapping/policy_mapping.md

```diff
@@ -0,0 +1,77 @@
+# Policy Mapping Workflow with IP/MAC
+
+This document describes the proposed changes for policy mapping workflow with IPs/MACs.
+
+## Background
+
+Currently, we have two ways to determine which policy gets applied to which workload:
+
+- Provide the VNF security manager with encapsulation tag
+- Provide the VNF security manager with IP/MAC
+
+If the manager is supporting [Policy Mapping](https://github.com/opensecuritycontroller/security-mgr-api/blob/master/src/main/java/org/osc/sdk/manager/api/ApplianceManagerApi.java#L40) and [Security Group](https://github.com/opensecuritycontroller/security-mgr-api/blob/master/src/main/java/org/osc/sdk/manager/api/ApplianceManagerApi.java#L41), we provide the manager with the security group name and members (their IPs and MACs) information to bind the policy.
+
+## Constraints and Assumptions
+The appliance security manager will not delete a security group with one or more security group interfaces.
+
+## Design Changes
+`SecurityGroup`: Currently, `SecurityGroup` and `SecurityGroupInterface` have many to many entity mapping. The proposed design will change this relationship to a one to many relationship between the SG and the SGIs
+
+### REST API
+`SecurityGroupInterfaceDto`:
+
+```
+[
+  {
+    "virtualSystemId": 0,
+    "name": "string",
+    "policyId": 0,
+    "mgrSGId": "string", // This new property is being added to store the security group id returned by the manager
+    "failurePolicyType": "FAIL_OPEN",
+    "order": 0,
+    "policies": [
+      {
+        "id": 0,
+        "parentId": 0,
+        "policyName": "string",
+        "mgrPolicyId": "string",
+        "mgrDomainId": 0,
+        "mgrDomainName": "string"
+      }
+    ],
+    "markedForDeletion": false,
+    "binded": false
+  }
+]
+```
+
+### OSC SDKs
+
+#### VNF Security Manager SDK
+
+`SecurityGroupInterfaceElement`: Create package org.osc.sdk.manager.input and add the `SecurityGroupInterfaceElement` interface with `name`, `mgrSGId` and `tag` of type string.
+
+`ManagerSecurityGroupInterfaceApi`: Add `SecurityGroupInterfaceElement` as a paramater to interface methods `createSecurityGroupInterface` and `updateSecurityGroupInterface`
+
+#### SDN Controller SDK
+No changes to SDN Controller SDK are required.
+
+### XYZ Integration Point
+The change to the VNF Security Manager SDK will break the plugin implementations. The plugin authors are required to update their plugins.
```

Comment:
Please change: The plugin authors are required to update their plugins -> Updating the plugins implementing these APIs is out of scope for this feature, with the exception of the `security-mgr-sample-plugin` which will be modified accordingly. 
With this you can remove the ln 62.